### PR TITLE
Present bookings when person information is unavailable

### PR DESCRIPTION
**Pull request by danlivings-dxw**
> See [ticket #1433 on the CAS3 Trello board](https://trello.com/c/vZ1cG8c0/1433-the-api-should-not-remove-a-booking-from-the-list-if-we-have-any-problem-fetching-data-spike).

This PR allows bookings to be returned by the `GET /premises/{premisesId}/bookings` and `GET /premises/{premisesId}/bookings/{bookingId}` endpoints when information about the person for whom the booking has been made is unavailable. It introduces a new `UnknownPerson` schema in the API specification, which allows consumers of the API to distinguish between a situation where the information is being intentionally withheld (represented by `RestrictedPerson`) and where the information is not currently available for some other reason (represented by `UnknownPerson`).

This PR depends on #804, and has the branch `309-replace-person-with-personinfo` set as the base to clarify the changes made for this PR. When that PR has been merged this will be updated to make `main` the base.

### CODE REVIEW

**Review by nsmj63-dxw — APPROVED** [review]
This one's ready to fire as soon as #804's been merged in 👍

**Comment by tahb** [comment]
Plan is to merge 804 in tomorrow when Stu returns. Rich has progressed the CAS1 UI work into prod to unlock this.

**Review by pezholio — APPROVED** [review]
LGTM - let's get it in!

**Comment by tahb** [comment]
We've merged 804 in, reverted it and finally [reproposed it](https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/914). Might be worth waiting for the latest PR to be merged before rebasing this one on `main`.

**Pull request merged**
